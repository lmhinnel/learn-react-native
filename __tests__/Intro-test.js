import React from 'react';
import renderer from 'react-test-renderer';
import Intro from '../src/utils/Intro';
import { FlatList, Text } from 'react-native';

test('renders correctly', () => {
  const tree = renderer.create(<Intro />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the flatlist component', () => {
  const tree = renderer.create(
    <FlatList
      data={['item1', 'item', 'item3']}
      keyExtractor={item => item}
      renderItem={({ item }) => <Text>{item}</Text>}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})