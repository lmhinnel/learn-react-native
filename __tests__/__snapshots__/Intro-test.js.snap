// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    This is a React Native snapshot test.
  </Text>
</View>
`;

exports[`renders the flatlist component 1`] = `
<RCTScrollView
  data={
    Array [
      "item1",
      "item",
      "item3",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        item1
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        item
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        item3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
